core: replace Cloudflare 403 HTML with friendly message

smalyu opened the pull request:
### Motivation

When Codex is launched from a region where Cloudflare blocks access (for example, Russia), the CLI currently dumps Cloudflare’s entire HTML error page. This isn’t actionable and makes it hard for users to understand what happened. We want to detect the Cloudflare block and show a concise, user-friendly explanation instead.

### What Changed

- Added CLOUDFLARE_BLOCKED_MESSAGE and a friendly_message() helper to UnexpectedResponseError. Whenever we see a 403 whose body contains the Cloudflare block notice, we now emit a single-line message (Access blocked by Cloudflare…) while preserving the HTTP status and request id. All other responses keep the original behaviour.
- Added two focused unit tests:
  - unexpected_status_cloudflare_html_is_simplified ensures the Cloudflare HTML case yields the friendly message.
  - unexpected_status_non_html_is_unchanged confirms plain-text 403s still return the raw body.

### Testing

- cargo build -p codex-cli
- cargo test -p codex-core
- just fix -p codex-core
- cargo test --all-features

smalyu committed:
core: show friendly message for Cloudflare 403 blocks

smalyu committed:
Fix formatting of Cloudflare error message in test

smalyu committed:
test: improve UnexpectedResponseError tests by deriving expected status from StatusCode

smalyu committed:
Merge branch 'main' into patch-2

smalyu committed:
fix(core): normalize request id punctuation

etraut-openai commented on codex-rs/core/src/error.rs:
```diff
@@ -235,18 +235,43 @@ pub struct UnexpectedResponseError {
     pub request_id: Option<String>,
 }
 
+const CLOUDFLARE_BLOCKED_MESSAGE: &str = "Access blocked by Cloudflare. This usually happens when connecting from a restricted region. Try again from a permitted network";
```
Please delete the phrase "Try again from a permitted network" at the end. The rest of the message is fine.

etraut-openai reviewed (changes requested):
Thanks for the contribution. The implementation looks fine. We have a small ask for the phrasing of the user message.

etraut-openai committed:
Merge branch 'main' into patch-2

smalyu committed:
Update Cloudflare blocked message

Removed the suggestion to try again from a permitted network in the Cloudflare blocked message.

smalyu committed:
style: reformat CLOUDFLARE_BLOCKED_MESSAGE declaration with cargo fmt
